### PR TITLE
Better GitHub releases

### PR DESCRIPTION
**Pull request by chshersh**
This PR is an attempt to bring better binary GitHub releases to `summoner`.

### CODE REVIEW

**Comment by chshersh** [comment]
Okay, so apparently, I can't uncomment macOS job easily. I see two possible solutions to proceed with this:

1. Use only `stack` on Travis CI to produce binaries for both mac and Linux.
2. Investigate [GitHub Actions releases](https://github.com/andys8/git-brunch/blob/master/.github/workflows/release.yml)

@vrom911 what do you think?

**Comment by vrom911** [comment]
@chshersh I don't mind having only Stack at one CI platform as long as we have the Cabal check somewhere else. I hope it will still produce good enough binaries :slightly_smiling_face: 

GitHub actions look promising, we can look into that, but only if it's not too time-consuming. In case of that, we can just go with the Stack-only option. What you personally prefer?

**Comment by chshersh** [comment]
@vrom911 I never used GitHub Actions for releases before. From what I saw, they don't look too difificult, but maybe there's a hidden catch. The problem is that to produce binaries, it's easier to use `stack` since `cabal` doesn't have CLI to get paths to produced binaries. But I don't know how to have `stack` on GitHub actions and have fast cached builds as well.

I'm also okay with using only `stack` at Travis CI, since we have `cabal` on GitHub actions 🙂 And AppVeyor CI also will have only `stack`...

**Comment by chshersh** [comment]
This PR should be ready to merge once CI passes. To highlight changes:

1. We now have better GitHub releases for all 3 operating systems. You can see example of such release here:
    * https://github.com/kowainik/summoner/releases/tag/v2.0-draft
    Any suggestions on names for files in there are appreciated 🙂 
2. We have robust Travis CI for both linux and macOS using `stack`.
3. AppVeyor was switched to `stack` as well.
4. All cabal example projects are tested properly with `cabal build` on GItHub actions.
5. All stack example projects are tested properly with `stack build` on Travis CI

File: appveyor.yml [hunk 1]
```diff
@@ -8,27 +7,51 @@ platform:
...
 
 environment:
-  global:
-    CABOPTS: --store-dir=C:\\SR
+  STACK_ROOT: C:\sr
+  STACK_VERSION: 2.1.1
+
+  # Workaround a gnarly bug https://github.com/haskell/cabal/issues/5386
+  # See: https://www.fpcomplete.com/blog/2018/06/sed-a-debugging-story
+  # TODO: check if it's fixed once we switch to lst-13 and GHC 8.6
```
Comment (vrom911) [inline appveyor.yml]: That might be an outdated comment :slightly_smiling_face:

Comment (chshersh) [reply appveyor.yml]: @vrom911 Good point! I'll remove it 👍

**Pull request merged**